src/axom/core/numerics/quadrature.hpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_NUMERICS_QUADRATURE_HPP_
+#define AXOM_NUMERICS_QUADRATURE_HPP_
+
+#include "axom/core/Array.hpp"
+
+/*!
+ * \file quadrature.hpp
+ * The functions declared in this header file find the nodes and weights of 
+ *   arbitrary order quadrature rules
+ */
+
+namespace axom
+{
+namespace numerics
+{
+
+/*!
+ * \class QuadratureRule
+ *
+ * \brief Stores a fixed array of 1D quadrature points and weights
+ */
+class QuadratureRule
+{
+  // Define friend functions so rules can only be created via compute_rule() methods
+  friend QuadratureRule compute_gauss_legendre(int npts);
+
+public:
+  QuadratureRule() = default;
+
+  //! \brief Accessor for quadrature nodes
+  double node(size_t idx) const { return m_nodes[idx]; };
```

Comment:
Design question -- do we always want this to be `double`? Or would we ever want other floating point types?

I think starting w/ double makes sense, but wanted to raise the question.